core/aggregations/src/main/java/com/microsoft/gctoolkit/aggregations/RuntimeAggregation.java

```diff
@@ -0,0 +1,75 @@
+package com.microsoft.gctoolkit.aggregations;
+
+import com.microsoft.gctoolkit.aggregator.Aggregation;
+import com.microsoft.gctoolkit.time.DateTimeStamp;
+
+/**
+ * An Aggregation that collates runtime data. This class is meant to be extended by other
+ * implementations that need access to runtime data to perform calculations. For example,
+ * to calculate the ratio of pause time to runtime duration.
+ */
+public abstract class RuntimeAggregation  implements Aggregation {
+
+    private volatile DateTimeStamp timeOfFirstEvent = null;
+    private volatile DateTimeStamp timeOfLastEvent = new DateTimeStamp(0d);
+
+    /**
+     * Runtime duration (in decimal seconds) after which we no longer consider the GC log to be a fragment
+     */
+    private static final double LOG_FRAGMENT_THRESHOLD = 18d;
+
+    /**
+     * This class is meant to be extended.
+     */
+    protected RuntimeAggregation() {}
+
+    /**
+     * RuntimeAggregation collates the time of an event and the duration of the event.
+     * @param eventTime The time a JVMEvent occurred.
+     * @param duration The duration of the JVMEvent.
+     */
+    public void record(DateTimeStamp eventTime, double duration) {
+        duration = !Double.isNaN(duration) ? duration : 0d;
+
+        if (timeOfFirstEvent == null || (eventTime != null && eventTime.before(timeOfFirstEvent))) {
```

Comment:
This will fail in the case of clock rollbacks. I would suggest that the first time a clock is seen, that is the start time of the first JVMEvent